src/core/lib/channel/channelz.cc

```diff
@@ -434,7 +436,11 @@ void PopulateSocketAddressJson(Json::Object* json, const char* name,
     if (!port.empty()) {
       port_num = atoi(port.data());
     }
-    char* b64_host = grpc_base64_encode(host.data(), host.size(), false, false);
+    grpc_resolved_address resolved_host;
+    grpc_string_to_sockaddr(&resolved_host, host.c_str(), port_num);
+    std::string packed_host = grpc_sockaddr_get_packed_host(&resolved_host);
+    char* b64_host = grpc_base64_encode(packed_host.data(), packed_host.size(),
```

Comment:
We probably shouldn't use `grpc_base64_encode` since it won't be able to deal with nul characters. Please use `absl::Base64Escape(packed_host)` instead. This will also return a `std::string` so we don't need to with `char*` either

Reply:
Based on a cursory reading of `grpc_base64_encode` and some experimentation with packed data including the `\0` character, this actually does work as is. Note that the function takes a size instead of just relying on null termination.

Regardless, switched over to `absl::Base64Escape`.

Reply:
The thing that confuses me is that it is not returning the size of the encoded string, so if the encoded string has nul characters, then it wouldn't work.

Reply:
Base64 won't include null characters, right? That's why it's useful for escaping.

Reply:
Oh, I didn't know that. Anyway, since we can use absl now, we should be nuking `grpc_base64_encode` :)